setup.py

```diff
@@ -1,70 +1,59 @@
-DOWNLOAD_URL        = 'https://github.com/ludwigschwardt/scikits.fitting'
-VERSION             = '0.5.1'
+# Ensure we have numpy before we start as it is needed before we call setup()
+# If not installed system-wide it will be downloaded into the local .eggs dir
+dist.Distribution(dict(setup_requires='numpy'))
 
-import os
-import setuptools
 from numpy.distutils.core import setup
+from numpy.distutils.misc_util import Configuration
+
 
 def configuration(parent_package='', top_path=None):
-    if os.path.exists('MANIFEST'): os.remove('MANIFEST')
+    if os.path.exists('MANIFEST'):
+        os.remove('MANIFEST')
 
-    from numpy.distutils.misc_util import Configuration
     config = Configuration(None, parent_package, top_path,
                            namespace_packages=['scikits'])
 
-    config.set_options(
-            ignore_setup_xxx_py=True,
-            assume_default_configuration=True,
-            delegate_options_to_subpackages=True,
-            quiet=True)
+    config.set_options(ignore_setup_xxx_py=True,
+                       assume_default_configuration=True,
+                       delegate_options_to_subpackages=True,
+                       quiet=True)
 
     config.add_subpackage('scikits')
-    config.add_subpackage(DISTNAME)
+    config.add_subpackage('scikits.fitting')
     config.add_data_files('scikits/__init__.py')
-    config.add_data_dir('scikits/fitting/tests')    
+    config.add_data_dir('scikits/fitting/tests')
 
     return config
 
-if __name__ == "__main__":
-
-    setup(
-        name=DISTNAME,
-        description=DESCRIPTION,
-        long_description=LONG_DESCRIPTION,
-        maintainer=MAINTAINER,
-        maintainer_email=MAINTAINER_EMAIL,
-        url=URL,
-        license=LICENSE,
-        download_url=DOWNLOAD_URL,
-        version=VERSION,
 
-        classifiers=['Development Status :: 4 - Beta',
-                     'Environment :: Console',
-                     'Intended Audience :: Developers',
-                     'Intended Audience :: Science/Research',
-                     'License :: OSI Approved :: BSD License',
-                     'Topic :: Scientific/Engineering'],
+if __name__ == "__main__":
 
-        configuration=configuration,
-        install_requires=[],
-        namespace_packages=['scikits'],
-        packages=setuptools.find_packages(),
-        include_package_data=True,
-        zip_safe=False, # the package can run out of an .egg file
-        #test_suite="tester", # for python setup.py test
-        )
+    with open('README.rst') as readme:
+        long_description = readme.read()
+
+    setup(name='scikits.fitting',
+          description='Framework for fitting functions to data with SciPy',
+          long_description=long_description,
+          maintainer='Ludwig Schwardt',
+          maintainer_email='ludwig@ska.ac.za',
+          url='https://github.com/ska-sa/scikits.fitting',
+          license='Modified BSD',
+          version='0.5.1',
```

Comment:
Indeed.

I'll need to patch katversion to handle scikits.fitting because of the dot in its name, otherwise I would have employed it already.